conf/turnserver.conf

```diff
@@ -2,10 +2,11 @@ lt-cred-mech
 use-auth-secret
 static-auth-secret=__TURNPWD__
 realm=__DOMAIN__
-no-stun
 
 tls-listening-port=__TLS_PORT__
 alt-tls-listening-port=__TLS_ALT_PORT__
+min-port=49152
+max-port=49192
```

Comment:
And what if one of these ports is already used by another app ?
Can't you reduce the range to 2 ports, or even only 1 ?

Reply:
> And what if one of these ports is already used by another app ?
It's not a problem if something else is a port in this range, coturn will just chose a other one for the connexion.

> Can't you reduce the range to 2 ports, or even only 1 ?
I think not. In the RFC the IP range is 49152 - 65535 (see https://tools.ietf.org/html/rfc5766#section-6.2). But we don't need so a lot. It's the reason that I have decided to use 40 port.

Reply:
Did you try with a smaller range ?
I'm not so fond of the idea of opening 40 ports without anyone behind it.
I would really prefer to use a few ports, reserved to synapse.

And, ok then, coturn will use another port if one is already used, but in your remove script you close all those ports.
So, again,
> And what if one of these ports is already used by another app ?

Reply:
I really understand your point of view.

It might work, but we can't guarantee that, because coturn will probably use one port per device. I didn't read the RFC but I think if there are too less port the connexion might fail when some device already use some port. I think we could try to reduce to 20 but less might make some problem sometime.

Reply:
I found [here](https://github.com/coturn/coturn/wiki/turnserver#options) that the default option is  49152 to 65535. I just can't imagine that you're suppose to open all these ports on your router.
Did you try without opening them ? coturn should probably used upnp to open the port dynamically as many app do.

You have to understand that keeping all those ports open is just unacceptable in regard of security matter.

Reply:
> Did you try without opening them ? coturn should probably used upnp to open the port dynamically as many app do.

Yes but it fail. We don't need so lot, but to have a full functional voip server we need to open some port. In all case upnp is not always functional.

>  You have to understand that keeping all those ports open is just unacceptable in regard of security matter.

It's the reason that I purpose to open 40 port. I've tried and it run successful.

Reply:
But 40 ports is huge ! That means 40 ports opened, and only one used at a time, so 39 other ports open without any services behind. It's too dangerous !

The only solution I can see is to open an issue on Synapse repository to ask how to manage that.

And, I would like to have other point of views on this matter. @YunoHost-Apps/apps-group

Reply:
For me, opening 40 ports on a rooter is way too much. We can't to that as it can cause problems of security (bypassing firewall). I agree with @maniackcrudelis : the only solution is to open an issue on Synapse Github repo.

Reply:
I think this issue should go more here https://github.com/coturn/coturn

We could try to reduce this to maybe 10 but after the problem might if a user try to do many call in the same time and some service already use some port.

Reply:
Well, yes... but in fact, we need to open them just in time. Why not opening them when a users want it ?

Reply:
Because coturn don't implement any feature to mange the firewall when he need to use a specific port.

Reply:
It's also a possibility but the problem is that the modification of the coturn config will be increased in each update.

Reply:
That sounds like a reasonable option to me, provided we can express the limitations without that ports open.

Reply:
And what does we set in coturn config 40 port ? or 10 ?

Reply:
I've made some new test. And with my last test I saw that with less thant 40 ports we get some instability. By example sometime the connexion success, sometime not, and sometime it's only on one direction. :-)